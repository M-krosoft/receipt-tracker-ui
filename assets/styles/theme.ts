import { createTheme } from "@shopify/restyle";

const palette = {
  darkBlue: "#253659",
  cyan: "#03A696",
  lightCyan: "#04BF9D",
  orange: "#F28157",
  darkOrange: "#BF665E",
  white: "#fffafa",
  transparent: "#00000000",
};

const theme = createTheme({
  colors: {
    mainBackgroundGradientFirst: palette.lightCyan,
    mainBackgroundGradientSecond: palette.cyan,

    textPrimary: palette.white,
    textAccent: palette.orange,

    transparent: palette.transparent,

    buttonPrimary: palette.white,
    buttonPrimaryForeground: palette.orange,

    buttonSecondary: palette.white,
    buttonSecondaryForeground: palette.darkOrange,
  },
  spacing: {
    auto: "auto",
    xs: 4,
    s: 8,
    m: 16,
    l: 24,
    xl: 40,
    xxl: 56,
    xxxl: 64,
  },
  borderRadii: {
    xs: 2,
    s: 4,
    m: 8,
    l: 16,
    xl: 24,
  },

  textVariants: {
    header1: {
      fontWeight: "bold",
      fontSize: 34,
    },
    header2: {
      fontWeight: "bold",
      fontSize: 28,
    },
    header3: {
      fontWeight: "bold",
      fontSize: 22,
    },
    body1: {
      fontSize: 16,
    },
    body2: {
      fontSize: 14,
    },
    body3: {
      fontSize: 12,
    },
    buttonPrimary: {
      backgroundColor: "transparent",
      fontWeight: "bold",
      borderRadius: "m",
      color: "buttonPrimary",
      paddingHorizontal: "m",
      paddingVertical: "s",
    },
    buttonSecondary: {
      fontWeight: "bold",
      borderRadius: "m",
      color: "buttonPrimary",
      padding: "s",
    },
    buttonText: {
      fontWeight: "bold",
    },
    defaults: {},
  },

  cardVariants: {
    screenContainer: {
      padding: "l",
    },
    section: {
      padding: "m",
    },
    card: {
      padding: "s",
    },

    defaults: {},
  },

  buttonVariants: {
    buttonPrimary: {
      borderRadius: "m",
      backgroundColor: "buttonPrimaryForeground",
    },

    buttonSecondary: {
      borderRadius: "m",
      backgroundColor: "buttonSecondaryForeground",
    },
    buttonText: {},
    defaults: {},
  },
});

export type Theme = typeof theme;
export default theme;
