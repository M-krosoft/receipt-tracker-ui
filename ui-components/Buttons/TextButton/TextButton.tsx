import { Theme } from "@assets/styles/theme";
import {
  color,
  composeRestyleFunctions,
  TextProps,
  typography,
  useRestyle,
} from "@shopify/restyle";
import { TouchableOpacity } from "react-native";
import { Text } from "@ui-components/Text/Text";

type RestyleProps = TextProps<Theme>;

const restyleTextFunctions = composeRestyleFunctions<Theme, RestyleProps>([
  color,
  typography,
]);

type Props = TextProps<Theme> & {
  onPress: () => void;
  label: string;
};

export const TextButton = ({ onPress, label, ...rest }: Props) => {
  const props = useRestyle(restyleTextFunctions, rest);
  return (
    <TouchableOpacity
      activeOpacity={0.9}
      onPress={onPress}
      accessibilityRole={"button"}
    >
      <Text fontWeight={"bold"} textAlign="center" {...props}>
        {label}
      </Text>
    </TouchableOpacity>
  );
};
