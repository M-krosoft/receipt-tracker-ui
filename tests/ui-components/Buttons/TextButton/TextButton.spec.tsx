import { TextButton } from "@ui-components/Buttons/TextButton/TextButton";
import theme from "@assets/styles/theme";
import { ThemeProvider } from "@shopify/restyle";
import { fireEvent, render } from "@testing-library/react-native";

describe("TextButton", () => {
  test("should render text button with correct styles", async () => {
    // When
    const renderedComponent = await render(
      <ThemeProvider theme={theme}>
        <TextButton onPress={jest.fn()} label="Login" fontSize={18} />
      </ThemeProvider>
    );

    // Then
    expect(renderedComponent.toJSON()).toMatchSnapshot();
  });

  test("should call onPress when text button is pressed", async () => {
    // Given
    const onPress = jest.fn();
    const renderedComponent = await render(
      <ThemeProvider theme={theme}>
        <TextButton onPress={onPress} label="Login" />
      </ThemeProvider>
    );

    // When
    const button = renderedComponent.getByText("Login");
    fireEvent(button, "press");

    // Then
    expect(onPress).toHaveBeenCalled();
  });
});
