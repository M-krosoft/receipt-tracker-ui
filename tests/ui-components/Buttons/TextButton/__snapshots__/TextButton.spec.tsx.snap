// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton should render text button with correct styles 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "fontWeight": "bold",
          "textAlign": "center",
        },
        [
          {
            "fontSize": 18,
          },
        ],
      ]
    }
  >
    Login
  </Text>
</View>
`;
